### PR TITLE
Add ember-template-lint to the syntax checkers

### PR DESCRIPTION
**Pull request by xcambar**
See https://github.com/ember-template-lint/ember-template-lint

It has been introduced in the default toolset of Ember in the latest releases (ember-cli 3.5, IIRC).

---

heavily inspired by https://pastebin.com/8iBaMrpj

### CODE REVIEW

**Comment by fmdkdd** [comment]
Thanks!  Can you add a test to `flycheck-test.el` as well?

**Comment by xcambar** [comment]
Tests have been added.

Unfortunately, this PR for ember-template-lint can not be merged as is, as it requires a number of modifications to be merged and published in the linter first:
* https://github.com/ember-template-lint/ember-template-lint/pull/580
* https://github.com/ember-template-lint/ember-template-lint/pull/582
* https://github.com/ember-template-lint/ember-template-lint/pull/586

**Comment by fmdkdd** [comment]
The CI is green but I'm still seeing one unmerged PR upstream for stdin.  Do we need to wait for it?

**Comment by xcambar** [comment]
I'll double triple check

File: flycheck.el [hunk 1]
```diff
@@ -7361,6 +7362,49 @@ The checker runs `checkdoc-current-buffer'."
...
+    ember-template
+    ".template-lintrc.js"
+  :type 'string
+  :safe #'stringp)
+
+(flycheck-define-checker ember-template
+  "An Ember template checker using ember-template-lint."
+  :command ("ember-template-lint"
+            (config-file "--config-path" flycheck-ember-template-lintrc)
+            "--stdin-filename" source-original
```
Comment (nag5000) [inline flycheck.el]: @fmdkdd @xcambar https://github.com/ember-template-lint/ember-template-lint/pull/783 is merged, so this works:

```suggestion
            "--filename" source-original
```

**Comment by xcambar** [comment]
Thanks @nag5000

**Comment by nag5000** [comment]
@fmdkdd It works well, can we merge this PR?

**Comment by fmdkdd** [comment]
Thanks for the ping.  LGTM.

**Pull request merged**